core/services/keeper/orm.go

```diff
@@ -99,8 +110,19 @@ DELETE FROM upkeep_registrations WHERE registry_id IN (
 	return rowsAffected, nil
 }
 
-func (korm ORM) EligibleUpkeepsForRegistry(registryAddress ethkey.EIP55Address, blockNumber, gracePeriod int64) (upkeeps []UpkeepRegistration, err error) {
-	stmt := `
+func (korm ORM) EligibleUpkeepsForRegistry(registryAddress ethkey.EIP55Address, head *types.Head, gracePeriod int64) (upkeeps []UpkeepRegistration, err error) {
+	registry, err := korm.RegistryByContractAddress(registryAddress)
+	if err != nil {
+		return nil, errors.Wrap(err, "EligibleUpkeepsForRegistry failed to get a registry by address")
+	}
+	blockNumber := head.Number
+	binaryHash, err := binaryOfFirstHashInTurn(blockNumber, registry, head)
+	if err != nil {
+		return nil, errors.Wrap(err, "EligibleUpkeepsForRegistry failed to convert hash to binary")
+	}
+
+	err = korm.q.Transaction(func(tx pg.Queryer) error {
```

Comment:
Why use transaction here?

Reply:
was there previously. i dont think we need it but leaving for now and will look into removing it separately

Reply:
I think you might be on an old version. I removed it in https://github.com/smartcontractkit/chainlink/pull/6229

Reply:
oh rebase/merge must have got messy. fixed